test/framework/control_plane.go

```diff
@@ -34,84 +34,191 @@ import (
 	"sigs.k8s.io/controller-runtime/pkg/client"
 )
 
-const (
-	// eventuallyInterval is the polling interval used by gomega's Eventually
-	eventuallyInterval = 10 * time.Second
+// Package globals used to populate default values.
+// Can be updated by consumer to change default values.
+var (
+	// DefaultTimeout is used for the default timeout for every `Eventually` call.
+	DefaultTimeout = 10 * time.Minute
```

Comment:
Hi @chuckha,

So gomega actually has a way to [assign default timeouts and polling intervals](https://onsi.github.io/gomega/#modifying-default-intervals). I usually don't opt to use them as it's process-wide. But since this is effectively doing that, maybe use those instead?

Reply:
the benefit of what this PR does is that each test can customize the poll interval and timeout. Using the built in assign in gomega wouldn't allow for that. If one test may take 5 minutes and one test should only take 10 seconds it would be nice to scope the timeouts to be that expectation, but this might be an over-optimization.

The weird part about what this PR is doing is that users familiar with gomega may set the defaults you linked to but they would have no effect.

Do you think this PR over optimizes for user customizability and it should use the defaults instead reducing what a user can do but integrating with gomega better?

Reply:
I suspect what the best default to use will vary based on the environment being used anyway. I think I'd be good with avoiding the defaults here and requiring them to be passed in by users...

Reply:
This was updated to use remove the Duration struct and integrate with Ginkgo more seamlessly